docs/docs/build-caching.md

```diff
@@ -3,7 +3,67 @@ title: Build Caching
 issue: https://github.com/gatsbyjs/gatsby/issues/8103
 ---
 
-This is a stub. Help our community expand it.
+Plugins can cache data as json objects and retrieve them on consecutive builds.
 
-Please use the [Gatsby Style Guide](/contributing/gatsby-style-guide/) to ensure your
-pull request gets accepted.
+Data is stored under the `/.cache/caches/{plugin-name}/` directory.
+
+
+## The cache api
+
+The cache api is accessible within the arguments object past to [Gatsby's apis](/docs/node-apis/).
+```
+exports.onPostBootstrap = async function({ cache, store, graphql }) {
+```
+
+
+The two functions you would want to use are:
+
+### SET
+`cache.set(key: String, obj: Object): Promise<any>`
+
+### GET
+`cache.get(key: String): Promise<Object>`
+
+
+
+## Custom plugin – example
+
+In your plugin's `gatsby-node.js` file, on the [Gatsby api](/docs/node-apis/) you wish to utilize cache in, include `cache` as an argument; eg.
+
+```
+exports.onPostBuild = async function(
+  { cache, store, graphql },
+  { query } ) {
+    const cacheKey = 'some-key-name';
+    let obj = await cache.get(cacheKey);
+
+    if (!obj) {
+      obj = { created: Date.now() };      
+      const data = await graphql(query);
+      obj.data = data;
+    } else if (Date.now() > obj.lastChecked + 3600000) {
+      /* Reload after a day */
+      const data = await graphql(query);
+      obj.data = data;
+    }
+    
+    obj.lastChecked = Date.now();    
+
+    await cache.set(cacheKey, obj);
+    
+    /* Do something with data ... */
+  }
+```
+
+**Note:** Adding an option such as `enableCaching` to your plugin's option could be used to toggle between using caching or not.
+
+
+## Clearing cache
+
+Since cache files are stored within the `/.cache/` directory, simply deleting it will clear all cache.
+Otherwise it will also be cleared when there are changes made to you `gatsby-config.js` file
+
+
+## Using with Netlify
```

Comment:
Let's remove this - we've seen a few strange-ish issues and it's generally a premature optimization.

It _can_ help your build, but it can also introduce some strange issues.

Reply:
@DSchau I actually have been struggling with cache on Netlify, and now that you mention this I am wondering if the "strange-ish" issues you talk of is what I am experiencing. Any ideas on why it doesn't work as expected?